jestSupport/preprocessor.js

```diff
@@ -26,6 +26,6 @@ module.exports = {
   getCacheKey: createCacheKeyFunction([
     __filename,
     path.join(__dirname, '../packager/transformer.js'),
-    path.join(__dirname, '../node_modules/babel-core/package.json'),
+    path.join(require.resolve('babel-core')),
```

Comment:
@Bhullnatik you can simple write `require.resolve('babel-core/package.json')` it works fine for npm@2.x and npm@3.x